lib/services/project-service.ts

```diff
@@ -68,25 +67,69 @@ export class ProjectService implements IProjectService {
 				this.$logger.trace("Copying custom app into %s", appDirectory);
 				appPath = customAppPath;
 			} else {
-				// No custom app - use nativescript hello world application
-				this.$logger.trace("Using NativeScript hello world application");
-				let defaultTemplatePath = this.$projectTemplatesService.defaultTemplatePath.wait();
-				this.$logger.trace("Copying NativeScript hello world application into %s", appDirectory);
+				let defaultTemplatePath = this.$projectTemplatesService.prepareTemplate(selectedTemplate).wait();
+				this.$logger.trace(`Copying application from '${defaultTemplatePath}' into '${appDirectory}'.`);
 				appPath = defaultTemplatePath;
 			}
 
 			try {
 				this.createProjectCore(projectDir, appPath, projectId).wait();
+				//update dependencies and devDependencies of newly created project with data from template
+				this.mergeProjectAndTemplateProperties(projectDir, appPath).wait();
+				this.updateAppResourcesDir(appDirectory).wait();
 			} catch (err) {
 				this.$fs.deleteDirectory(projectDir).wait();
 				throw err;
 			}
-
 			this.$logger.out("Project %s was successfully created", projectName);
 
 		}).future<void>()();
 	}
 
+	private mergeProjectAndTemplateProperties(projectDir: string, templatePath: string): IFuture<any> {
+		return ((): any => {
+			let projectPackageJsonPath = path.join(projectDir, constants.PACKAGE_JSON_FILE_NAME);
+			let projectPackageJsonData = this.$fs.readJson(projectPackageJsonPath).wait();
+			this.$logger.trace("Initial project package.json data: ", projectPackageJsonData);
+			let templatePackageJsonData = this.$fs.readJson(path.join(templatePath, constants.PACKAGE_JSON_FILE_NAME)).wait();
+			if(projectPackageJsonData.dependencies || templatePackageJsonData.dependencies) {
+				projectPackageJsonData.dependencies = this.mergeDependencies(projectPackageJsonData.dependencies, templatePackageJsonData.dependencies);
+			}
+
+			if(projectPackageJsonData.devDependencies || templatePackageJsonData.devDependencies) {
+				projectPackageJsonData.devDependencies = this.mergeDependencies(projectPackageJsonData.devDependencies, templatePackageJsonData.devDependencies);
+			}
+
+			this.$logger.trace("New project package.json data: ", projectPackageJsonData);
+			this.$fs.writeJson(projectPackageJsonPath, projectPackageJsonData).wait();
+		}).future<any>()();
+	}
+
+	private updateAppResourcesDir(appDirectory: string): IFuture<any> {
+		return ((): any => {
+			let defaultAppResourcesDir = path.join(this.$projectTemplatesService.defaultTemplatePath.wait(), constants.APP_RESOURCES_FOLDER_NAME);
+			let targetAppResourcesDir = path.join(appDirectory, constants.APP_RESOURCES_FOLDER_NAME);
+			this.$logger.trace(`Updating AppResources values from ${defaultAppResourcesDir} to ${targetAppResourcesDir}`);
+			(<any>shelljs).config.silent = true;
```

Comment:
Why are we not using https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/shelljs/shelljs.d.ts ?